list.cpp

```diff
@@ -0,0 +1,145 @@
+#include "list.h"
+#include <iostream>
+using std::cout;
+using std::endl;
+
+
+List::List(){
+	head = nullptr;
+	node = nullptr;
+	curr1 = nullptr;
+	curr2 = nullptr;
+}
+
+void List::addNode(int idata, int option){
+	node = new Node;	//create a node using dynamic memory allocation
+	node->data = idata;
+
+	//check if the node just created is a first node
+	if (head == nullptr){
+		head = node;
+		curr1 = node;
+		curr2 = node;
+		node->next = nullptr;
+	}
+	else{
+		//This means if we already have first node then we'll do something else
+
```

Comment:
the use of options below is interesting, and not at all a bad idea :-).  it's not quite as general as using a normalized index, but it retains much of the utility (since the most common operations, in many use cases, will be to insert and remove from the beginning or end of the list) -- and i'm not sure that it simplifies things significantly in actuality, but it certainly makes the code feel simpler.